src/databricks/labs/ucx/code/queries.py

```diff
@@ -0,0 +1,59 @@
+from typing import Iterable
+
+import sqlglot
+from sqlglot.expressions import Table
+
+from databricks.labs.ucx.code.base import Fixer, Diagnostic, Position, Range, Severity, Linter, DiagnosticTag
+
+from databricks.labs.ucx.hive_metastore.table_migrate import Index
+
+
+class FromTable(Linter, Fixer):
+    def __init__(self, index: Index):
+        self._index = index
+
+    def lint(self, query: str) -> Iterable[Diagnostic]:
+        for statement in sqlglot.parse(query):
+            if not statement:
+                continue
+            for table in statement.find_all(Table):
+                catalog = self._catalog(table)
+                if catalog != 'hive_metastore':
+                    continue
+                if not self._index.is_upgraded(table.db, table.name):
+                    continue
+                dst = self._index.get(table.db, table.name)
```

Comment:
is_upgraded and get method doing iteration two times, first to check whether it's upgraded and second time to get it.  Rather, it can be combined to just a single method by using get method only. If we receive other than None from get method, we proceed else skip loop.